cmd/gitcloner/integrationtests/clone_test.go

```diff
@@ -0,0 +1,465 @@
+		})
+
+		JustAfterEach(func() {
+			err := os.RemoveAll(tmp)
+			Expect(err).NotTo(HaveOccurred())
+		})
+
+		When("Private repo that contains a README.md file", func() {
+			BeforeEach(func() {
+				private = true
+				opts = &cmd.Options{
+					InsecureSkipTLS: true,
+				}
+			})
+
+			AfterEach(func() {
+				err := os.RemoveAll(tmpKey)
+				Expect(err).NotTo(HaveOccurred())
+			})
+
+			JustBeforeEach(func() {
+				c := gogit.NewCloner()
+				cloneErr = c.CloneRepo(opts)
+				Expect(cloneErr).NotTo(HaveOccurred())
+			})
+
+			It("README.md file is present", func() {
+				_, err := os.Stat(tmp + "/README.md")
+				Expect(err).NotTo(HaveOccurred())
+			})
+		})
+	})
+})
+
+// createGogsContainer creates a container that runs gogs. It creates a certificate for tls, and stores the ca bundle in gogsCABundle
+func createGogsContainer() (testcontainers.Container, error) {
+	tmpDir := createTempFolder()
+	err := cp.Copy("assets/gitserver", tmpDir)
+	if err != nil {
+		return nil, err
+	}
+	req := testcontainers.ContainerRequest{
+		Image:        "gogs/gogs:0.13",
+		ExposedPorts: []string{"3000/tcp", "22/tcp"},
+		Mounts: testcontainers.ContainerMounts{
+			{
+				Source: testcontainers.GenericBindMountSource{HostPath: tmpDir},
+				Target: "/data",
+			},
+		},
+	}
+	container, err := testcontainers.GenericContainer(context.Background(), testcontainers.GenericContainerRequest{
+		ContainerRequest: req,
+		Started:          true,
+	})
+
+	if err != nil {
+		return nil, err
+	}
+
+	// create ca bundle
+	_, err = container.Exec(context.Background(), []string{"./gogs", "cert", "-ca=true", "-duration=8760h0m0s", "-host=localhost"})
+	if err != nil {
+		return nil, err
+	}
+	_, err = container.Exec(context.Background(), []string{"chown", "git:git", "cert.pem", "key.pem"})
+	if err != nil {
+		return nil, err
+	}
+	caReader, err := container.CopyFileFromContainer(context.Background(), "/app/gogs/cert.pem")
+	if err != nil {
+		return nil, err
+	}
+	gogsCABundle, err = io.ReadAll(caReader)
+	if err != nil {
+		return nil, err
+	}
+
+	url, err := getHTTPSURL(context.Background(), container)
+	if err != nil {
+		return nil, err
+	}
+
+	c := gogs.NewClient(url, "")
+	//nolint:gosec // need insecure TLS option for testing
+	tr := &http.Transport{
+		TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+	}
+	httpClient := &http.Client{Transport: tr} // #nosec G402
+	c.SetHTTPClient(httpClient)
+
+	// create access token, we need to wait until the http server is available
+	Eventually(func() error {
+		token, err := c.CreateAccessToken(gogsUser, gogsPass, gogs.CreateAccessTokenOption{
+			Name: "test",
+		})
+		if err != nil {
+			return err
+		}
+		gogsClient = gogs.NewClient(url, token.Sha1)
+		gogsClient.SetHTTPClient(httpClient)
+
+		return nil
+	}, "5s", "200ms").ShouldNot(HaveOccurred())
+
+	return container, nil
+}
+
+// createRepo creates a git repo for testing
+func createRepo(url string, path string, private bool) error {
+	name := "test-repo"
```

Comment:
This value is also used in test cases when computing repo URLs; I think that this should perhaps be a suite-wide constant, or a parameter passed to this function. How do you see it?

Reply:
done in https://github.com/rancher/gitjob/pull/302/commits/43c66e7e40a9791d03bff8bc45172e4d1e87742e